opm/simulators/wells/BlackoilWellModelGeneric.cpp

```diff
@@ -996,7 +996,17 @@ checkGroupProductionConstraints(const Group& group,
             // sum over all nodes
             current_rate = comm_.sum(current_rate);
 
-            if (controls.liquid_target < current_rate  ) {
+            bool skip = false;
+            if (controls.liquid_target == controls.oil_target) {
+                double current_water_rate = WellGroupHelpers::sumWellSurfaceRates(group, schedule(), well_state, reportStepIdx, phase_usage_.phase_pos[BlackoilPhases::Aqua], false);
+                current_water_rate = comm_.sum(current_water_rate);
```

Comment:
Have we not communicated and stored the group rates in the well state, is it necessary to do that again here?